Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -104,6 +104,24 @@ private PowerShell CreatePowerShell()
                 PSModulePathEnvName,
                 $"{psJupyterModulePath}{Path.PathSeparator}{psModulePath}");
 
+            // Set $PROFILE.
+            string allUsersCurrentHost = DollarProfileHelper.GetFullProfileFilePath(forCurrentUser: false);
+            string currentUserCurrentHost = DollarProfileHelper.GetFullProfileFilePath(forCurrentUser: true);
+            PSObject dollarProfile = DollarProfileHelper.GetDollarProfile(allUsersCurrentHost, currentUserCurrentHost);
+
+            pwsh.Runspace.SessionStateProxy.SetVariable("PROFILE", dollarProfile);
+
+            // Run the PROFILE scripts if they exist.
+            if (File.Exists(allUsersCurrentHost))
+            {
+                pwsh.AddScript(allUsersCurrentHost).InvokeAndClearCommands();
+            }
+
+            if (File.Exists(currentUserCurrentHost))
+            {
+                pwsh.AddScript(currentUserCurrentHost).InvokeAndClearCommands();
+            }
```

Comment:
How to handle the errors generated by profile execution may be a problem.
You can use the code at https://github.com/dotnet/interactive/pull/344/files#diff-8ddc46fdba4f3409e381fdab7b1fd6edR275 but I'm not sure if the output written to stdout/stderr will be rendered (reliably) to the frontend at this stage because dotnet-interactive only capture stdout/stderr when it's in a `code submit` event.

Reply:
that link doesn't take me anywhere - can you supply another

Reply:
I've done some refactoring.